eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/ProxyConfigurationTest.java

```diff
@@ -0,0 +1,114 @@
+package com.azure.eventhubs;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.function.Consumer;
+
+import static com.azure.eventhubs.ProxyConfiguration.SYSTEM_DEFAULTS;
+
+@RunWith(Theories.class)
+public class ProxyConfigurationTest {
+
+    private static final String PROXY_HOST = "/127.0.0.1";
+    private static final String PROXY_PORT = "3128";
+    private static final String HTTP_PROXY = String.join(":", PROXY_HOST, PROXY_PORT);
+    private static final String PROXY_USERNAME = "dummyUsername";
+    private static final String PROXY_PASSWORD = "dummyPassword";
+
+    private static Proxy proxyAddress = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(PROXY_HOST, Integer.parseInt(PROXY_PORT)));
+
+    private static final Consumer<ProxyConfiguration> validateProxyConfiguration =  proxyConfiguration -> {
```

Comment:
The naming on this is probably causing a Checkstyle failure because it is not VALIDATE_PROXY_CONFIGURATION (ie. static final). This would be better as a method `void validateProxyConfiguration(ProxyConfiguration)`.